akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -36,7 +35,7 @@ import akka.util.ByteString
 import akka.util.JavaDurationConverters._
 import akka.util.unused
 
-object Tcp extends ExtensionId[Tcp] with TcpImplicitExtensionIdApply with ExtensionIdProvider {
+object Tcp extends ExtensionId[Tcp] with ExtensionIdProvider {
```

Comment:
Technically this is not binary compatible? Should we keep internal class empty around? Or hope that the internal class referenced accidentally?